scrapy_poet/page_input_providers.py

```diff
@@ -97,25 +105,25 @@ def __call__(self, to_provide, response: Response):
     provided_classes: ClassVar[Union[Set[Callable], Callable[[Callable], bool]]]
     name: ClassVar[str] = ""  # It must be a unique name. Used by the cache mechanism
 
-    @classmethod
-    def is_provided(cls, type_: Callable):
+    def is_provided(self, type_: Callable) -> bool:
         """
         Return ``True`` if the given type is provided by this provider based
         on the value of the attribute ``provided_classes``
         """
-        if isinstance(cls.provided_classes, Set):
-            return type_ in cls.provided_classes
-        elif callable(cls.provided_classes):
-            return cls.provided_classes(type_)
+        if isinstance(self.provided_classes, Set):
+            return type_ in self.provided_classes
+        elif callable(self.provided_classes):
+            return self.provided_classes(type_)
         else:
             raise MalformedProvidedClassesError(
-                f"Unexpected type '{type_}' for 'provided_classes' attribute of"
-                f"'{cls}.'. Expected either 'set' or 'callable'"
+                f"Unexpected type '{type_}' for 'provided_classes' attribute of "
+                f"'{self}.'. Expected either 'set' or 'callable'"
             )
 
-    def __init__(self, crawler: Crawler):
+    # FIXME: Can't import the Injector as class annotation due to circular dep.
```

Comment:
We're only using the `.build_instances()` method of the `Injector` here. Perhaps a strong sign that we need to decouple the instance creation and injection functionalities of `Injector`.